osu.Server.BeatmapSubmission/Services/S3BeatmapStorage.cs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using Amazon;
+using Amazon.Runtime;
+using Amazon.S3;
+using Amazon.S3.Model;
+using osu.Framework.Extensions;
+using osu.Game.IO.Archives;
+using osu.Server.BeatmapSubmission.Configuration;
+using osu.Server.BeatmapSubmission.Models;
+using SharpCompress.Writers;
+using SharpCompress.Writers.Zip;
+
+namespace osu.Server.BeatmapSubmission.Services
+{
+    public class S3BeatmapStorage : IBeatmapStorage
+    {
+        private const string osz_directory = "osz";
+        private const string versioned_file_directory = "beatmap_files";
+
+        private readonly AmazonS3Client client;
+
+        public S3BeatmapStorage()
+        {
+            client = new AmazonS3Client(
+                new BasicAWSCredentials(AppSettings.S3AccessKey, AppSettings.S3SecretKey),
+                new AmazonS3Config
+                {
+                    CacheHttpClient = true,
+                    HttpClientCacheSize = 32,
```

Comment:
I think this was tuned for performance a long time ago, but reading the internal comments I'm not sure this is working in our favour anymore?

![JetBrains Rider 2024-12-24 at 06 42 03](https://github.com/user-attachments/assets/c1c2a216-d2e5-4cae-bc1c-1a8628b8327e)

Might be worth removing this and seeing how things go?

Reply:
Removed in https://github.com/ppy/osu-server-beatmap-submission/pull/13/commits/c4674baa9c11df521fc35ec2c3845eec0893b4bf